sklearn/pipeline.py

```diff
@@ -37,6 +39,32 @@
 __all__ = ["Pipeline", "FeatureUnion", "make_pipeline", "make_union"]
 
 
+@contextmanager
+def _handle_warnings(estimator):
+    """A context manager to make sure a NotFittedError is raised, if a subestimator
+    raises the error.
+
+    Otherwise, we raise a warning if the pipeline is not fitted, with the deprecation.
+
+    TODO(1.8): remove this context manager and replace with check_is_fitted.
+    """
+    try:
+        yield
+    except NotFittedError:
+        check_is_fitted(estimator)
+
+    # we only get here if the above didn't raise
+    try:
+        check_is_fitted(estimator)
+    except NotFittedError:
+        warnings.warn(
+            "This Pipeline instance is not fitted yet. Call 'fit' with "
+            "appropriate arguments before using transform. This will raise "
```

Comment:
"using transform" -> I assume that it could another method. We might require to add the method name as an argument.